src/bluesearch/database/article.py

```diff
@@ -264,11 +266,80 @@ class JATSXMLParser(ArticleParser):
         The path to a JATS XML file.
     """
 
-    def __init__(self, path: str | Path) -> None:
+    def __init__(self, xml_stream: IO) -> None:
         super().__init__()
-        self.content = ElementTree.parse(str(path))
+        self.content = ElementTree.parse(xml_stream)
         self.ids = self.get_ids()
 
+    @classmethod
+    def from_string(cls, xml_string: str) -> JATSXMLParser:
+        """Read xml string and instantiate JATSXML Parser.
+
+        Parameters
+        ----------
+        xml_string
+            Raw content of the article
+
+        Returns
+        -------
+        JATSXMLParser
+            Parser containing the article content.
+        """
+        with StringIO(xml_string) as stream:
+            obj = cls(stream)
+        return obj
+
+    @classmethod
+    def from_xml(cls, path: str | Path) -> JATSXMLParser:
+        """Read xml file and instantiate JATSXML Parser.
+
+        Parameters
+        ----------
+        path
+            Path to the article (with .xml extension)
+
+        Returns
+        -------
+        JATSXMLParser
+            Parser containing the article content.
+        """
+        with open(path) as fh:
+            obj = cls(fh)
+        return obj
+
+    @classmethod
+    def from_zip(cls, path: str | Path) -> JATSXMLParser:
+        """Read xml file and instantiate JATSXML Parser.
```

Comment:
```suggestion
        """Read xml file from a zipped .meca folder and instantiate JATSXML Parser.
```
Just to avoid having the very same docstring for the two methods.